bundles/replication/src/main/java/com/adobe/cq/commerce/demandware/replication/transport/AbstractOCAPITransportPlugin.java

```diff
@@ -205,18 +200,22 @@ public boolean deliver(JSONObject delivery, AgentConfig config, ReplicationLog l
      *
      * @param method   the HTTP request method
      * @param delivery the JSON data
+     * @param dwInstanceId
      * @return the created request builder
      * @throws ReplicationException in case request builder can not be created
      */
-    protected RequestBuilder getRequestBuilder(final String method, final JSONObject delivery)
-            throws ReplicationException {
+    protected RequestBuilder getRequestBuilder(final String method, final JSONObject delivery,
+                                               final String dwInstanceId) throws ReplicationException {
         if (StringUtils.isEmpty(method)) {
             throw new ReplicationException("No request method provided");
         }
         try {
             // construct the OCAPI request
             final StringBuilder transportUriBuilder = new StringBuilder();
-            transportUriBuilder.append(DemandwareClient.DEFAULT_SCHEMA).append(demandwareClient.getEndpoint());
+            final String endpoint = clientProvider.getClientForSpecificInstance(dwInstanceId)
+                    .map(DemandwareClient::getEndpoint)
+                    .orElse(StringUtils.EMPTY);
+            transportUriBuilder.append(DemandwareClient.DEFAULT_SCHEMA).append(endpoint);
```

Comment:
is an empty endpoint really a valid option?

Reply:
Will change it to `throw new ReplicationException` in the client not found